### PR TITLE
Issue #659 Deal with issues for HFBs with unstructured grids.

### PR DESCRIPTION
**Pull request by JoerivanEngelen**
This PR resolves:

https://github.com/Deltares/imod-python/issues/697
https://github.com/Deltares/imod-python/issues/696

Work as part of:

https://github.com/Deltares/imod-python/issues/659

**This PR:**
- aligns the ``bounding_polygon`` function for unstructured grids with that of the structured grids: return bounding_polygon of active domain.
- catch edge case in when line crosses only vertex of bounding polygon, a point is returned. Drop these points.
- add tests with unstructured grid and HFB in various orientations

### CODE REVIEW

**Comment by JoerivanEngelen** [comment]
I checked why the pipeline was failing, but it was only due to some MyPy issues, this one is ready to be reviewed!

File: imod/mf6/multimodel/modelsplitter.py [hunk 1]
```diff
@@ -94,7 +94,7 @@ def slice_model(partition_info: PartitionInfo, model: Modflow6Model) -> Modflow6
         # to drop these packages. Create schemata dict only containing the
         # variables with a AllNoDataSchema.
         allnodata_schemata = filter_schemata_dict(
-            package._write_schemata, (AllNoDataSchema)
+            package._write_schemata, (AllNoDataSchema, IndexesSchema)
```
Comment (luitjansl) [inline imod/mf6/multimodel/modelsplitter.py:97]: update the comment above, now it only mentions AllNoDataSchema. The variable name allnodata_schemata can also be renamed appropriately

**Review by luitjansl — COMMENTED** [review]
looks good.

**Comment by Manangka** [comment]
Please start the title with "Issue 659". That way we get a nice and easy to read master branch 
(I would even be in favor of adding an # before the issue number)

File: docs/api/changelog.rst [hunk 2]
```diff
@@ -20,6 +20,13 @@ Fixed
   :function:`imod.select.points.point_values`
```
Comment (Manangka) [inline docs/api/changelog.rst:20]: Nice that you immediately update the changelog!
This should be part of our definition of done

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py [hunk 3]
```diff
@@ -216,6 +290,54 @@ def test_partitioning_unstructured_with_vpt_cells(
...
+@parametrize_with_cases("partition_array", cases=PartitionArrayCases)
+@parametrize_with_cases("hfb", cases=HorizontalFlowBarrierCases)
+def test_partitioning_unstructured_hfb(
+    tmp_path: Path,
+    circle_model: Modflow6Simulation,
+    partition_array: xu.UgridDataArray,
+    hfb: imod.mf6.HorizontalFlowBarrierBase,
+):
+    simulation = circle_model
+    # increase the recharge to make the head gradient more pronounced
```
Comment (Manangka) [inline imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:303]: Please start your comments with a captial. They should be as close to proper english as possible.

https://peps.python.org/pep-0008/#comments

> Comments should be complete sentences. The first word should be capitalized, unless it is an identifier that begins with a lower case letter (never alter the case of identifiers!).

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py [hunk 4]
```diff
@@ -216,6 +290,54 @@ def test_partitioning_unstructured_with_vpt_cells(
...
+    simulation["GWF_1"]["rch"]["rate"] *= 100
+
+    simulation["GWF_1"]["hfb"] = imod.mf6.HorizontalFlowBarrierResistance(geometry=hfb)
+
+    # run the original example, so without partitioning, and save the simulation results
+    orig_dir = tmp_path / "original"
```
Comment (Manangka) [inline imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:309]: Don't use abbreviations -> `original_dir`

File: imod/mf6/multimodel/modelsplitter.py (hunk 1 above)
Comment (JoerivanEngelen) [reply imod/mf6/multimodel/modelsplitter.py:97]: I added some comment, and refactored the crucial logic of IndexesSchema into the EmptyIndexesSchema, the latter object doesn't check for monotonicity, which is what is what is required here.

File: docs/api/changelog.rst (hunk 2 above)
Comment (JoerivanEngelen) [reply docs/api/changelog.rst:20]: Thanks, I think our definition of done should be added to the developer docs, right?

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py (hunk 3 above)
Comment (JoerivanEngelen) [reply imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:303]: Thanks, done!

File: imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py (hunk 4 above)
Comment (JoerivanEngelen) [reply imod/tests/test_mf6/test_multimodel/test_mf6_partitioning_unstructured.py:309]: Done

**Review by Manangka — APPROVED** [review]
